### DISCUSSION TITLE
Backups

### DISCUSSION THREAD

**Discussion by p-jones** [TandoorRecipes/recipes#2585] [General]
Thank you for writing this excellent software!

I am writing about backups. I have read the info [here](https://docs.tandoor.dev/system/backup/) and have tried two things:

**1) Export**
I have 581 recipes and the recipes appear to be exported. But when the process finishes, the page says the link has already expired so I cannot download the exported recipe zip file.

**2) Postgres DB dump**
I have tried using the command provided (`sudo docker exec -t docker_db_recipes_1 pg_dumpall -U djangouser > pgdump.sql`) but the file I get is only 136 bytes so I am unsure what is being backed up here. I have used the name of my Tandoor db container in place of `docker_db_recipes_1`.

**3) User comments/ratings**
I presume the postgres db backup, when it is working, will export not only the recipes, but also the user comments/ratings/etc? Just wanted to verify this.

I wondered if anyone had any tips? The software is excellent but in order for me to migrate completely to it, I will need a solid backup strategy.

Thank you!

**Comment by smilerz** [comment]
Exporting the database and backing up the location of the media files (tandoor container path/mediafiles by default) is a complete backup.

**Reply by p-jones** [reply]
I wonder why my export cannot be completed then. I never get a link to download the zip file (it says it has expired). Thanks.

**Reply by smilerz** [reply]
The export from tandoor is not an export of the database. It’s an export of recipes and is not intended to be a backup of the entire system.  There is an open issue describing the issues you have with export elsewhere. 

You can run a backup of the db manually with syntax similar to:
`docker exec -t your-db-container pg_dumpall -c -U postgres > dump_`date +%d-%m-%Y"_"%H_%M_%S`.sql`

I personally use the container `prodrigestivill/postgres-backup-local` to back up the db on a schedule.
